pkg/pss/pss_test.go

```diff
@@ -85,70 +69,70 @@ func TestTrojanChunkRetrieval(t *testing.T) {
 	}
 	storedChunk = tc.WithTagID(tag.Uid)
 
+	// check if receipt is received
+	if receipt == nil {
+		t.Fatal("receipt not received")
+	}
+
 	if !reflect.DeepEqual(tc, storedChunk) {
-		t.Fatalf("store chunk does not match sent chunk")
+		t.Fatalf("trojan chunk created does not match sent chunk. got %s, want %s", storedChunk.Address().ByteString(), tc.Address().ByteString())
 	}
 }
 
-// TestPssMonitor creates a trojan chunk
+// TestPssTags creates a trojan chunk
 // mocks the localstore
 // calls pss.Send method
 // updates the tag state (Stored/Sent/Synced)
 // waits for the monitor to notify the changed state
-func TestPssMonitor(t *testing.T) {
+func TestPssTags(t *testing.T) {
 	var err error
 	ctx := context.TODO()
 	testTags := tags.NewTags()
 
-	localStore := mock.NewTagsStorer(testTags)
-
 	target := trojan.Target([]byte{1}) // arbitrary test target
 	targets := trojan.Targets([]trojan.Target{target})
-	payload := []byte("RECOVERY CHUNK")
-	topic := trojan.NewTopic("RECOVERY TOPIC")
+	payload := []byte("PSS CHUNK")
+	topic := trojan.NewTopic("PSS TOPIC")
 
-	var monitor *pss.Monitor
+	pushSyncService := pushsyncmock.New(func(ctx context.Context, chunk swarm.Chunk) (*pushsync.Receipt, error) {
+		rcpt := &pushsync.Receipt{
+			Address: swarm.NewAddress(chunk.Address().Bytes()),
+		}
+		tt, err := testTags.Get(chunk.TagID())
+		if err != nil {
+			t.Fatal(err)
+		}
+		tt.Inc(tags.StateStored)
+		tt.Inc(tags.StateSent)
+		tt.Inc(tags.StateSynced)
+		return rcpt, nil
+	})
 
-	pss := pss.NewPss(localStore, testTags)
+	pss := pss.New(logging.New(ioutil.Discard, 0), pushSyncService, testTags)
 
+	var tag *tags.Tag
 	// call Send to store trojan chunk in localstore
-	if monitor, err = pss.Send(ctx, targets, topic, payload); err != nil {
+	if tag, err = pss.Send(ctx, targets, topic, payload); err != nil {
 		t.Fatal(err)
 	}
 
+	time.Sleep(1000 * time.Millisecond)
+
 	storeTags := testTags.All()
 	if len(storeTags) != 1 {
 		t.Fatalf("expected %d tags got %d", 1, len(storeTags))
 	}
 
-	timeout := 1 * time.Second
-	for _, expectedState := range []tags.State{tags.StateStored, tags.StateSent, tags.StateSynced} {
-		storeTags[0].Inc(expectedState)
-	loop:
-		for {
-			// waits until the monitor state has changed or timeouts
-			select {
-			case state := <-monitor.State:
-				if state == expectedState {
-					break loop
-				}
-			case <-time.After(timeout):
-				t.Fatalf("no message received")
-			}
-		}
+	if tag.Get(tags.StateStored) != 1 && tag.Get(tags.StateSent) != 1 && tag.Get(tags.StateSynced) != 1 {
+		t.Fatalf("Trojan Chunk expected to be Stored == %d, Sent == %d and Synced == %d", tag.Stored, tag.Sent, tag.Synced)
```

Comment:
why would trojan chunk be _stored_?

Reply:
This was a wrong assumption from when we ported.
Corrected, good point